File: DPP/21-06-2026/CPP/Q1.cpp
#include <iostream>
#include <vector>
#include <climits>
#include <cstdlib>

class Solution {
public:
    int closestPairToTarget(std::vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;

        int minDiff = INT_MAX;

        while (left < right) {
            int currentSum = nums[left] + nums[right];
            int absDiff = std::abs(currentSum - target);

            if (absDiff < minDiff) {
                minDiff = absDiff;
            }

            if (currentSum < target) {
                left++;
            } else if (currentSum > target) {
                right--;
            } else {
                break;
            }
        }

        return minDiff;
    }
};

int main() {
    Solution sol;

    std::vector<int> nums = {1, 3, 5, 8, 12};
    int target = 10;
    int result = sol.closestPairToTarget(nums, target);

    std::cout << result << std::endl;

    return 0;
}

File: DPP/21-06-2026/Python/Q1.py
from typing import List

class Solution:
    def closestPairToTarget(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        min_diff = float('inf')

        while left < right:
            current_sum = nums[left] + nums[right]
            abs_diff = abs(current_sum - target)

            if abs_diff < min_diff:
                min_diff = abs_diff
            
            if current_sum < target:
                left += 1
            elif current_sum > target:
                right -=1
            else:
                break
        
        return min_diff

sol = Solution()

nums = [1, 3, 5, 8, 12]
target = 10
result = sol.closestPairToTarget(nums, target)
print(result)